local_planner/src/nodes/planner_functions.cpp

```diff
@@ -45,13 +59,36 @@ void filterPointCloud(
     }
   }
 
-  cropped_cloud.header.stamp = complete_cloud[0].header.stamp;
-  cropped_cloud.header.frame_id = complete_cloud[0].header.frame_id;
-  cropped_cloud.height = 1;
-  cropped_cloud.width = cropped_cloud.points.size();
-  if (cropped_cloud.points.size() <= min_cloud_size) {
-    cropped_cloud.points.clear();
-    cropped_cloud.width = 0;
+  // combine with old cloud
+  for (const pcl::PointXYZI& xyzi : old_cloud) {
+    // adding older points if not expired and space is free according to new
+    // cloud
+    if (histogram_box.isPointWithinBox(xyzi.x, xyzi.y, xyzi.z)) {
+      distance = (position - toEigen(xyzi)).norm();
+      if (distance < histogram_box.radius_) {
+        PolarPoint p_pol = cartesianToPolar(toEigen(xyzi), position);
+        Eigen::Vector2i p_ind = polarToHistogramIndex(p_pol, ALPHA_RES / 2);
+        if (high_res_histogram.get_age(p_ind.y(), p_ind.x()) == 0 &&
+            xyzi.intensity < max_age) {
+          final_cloud.points.push_back(
+              toXYZI(xyzi.x, xyzi.y, xyzi.z, xyzi.intensity + 1));
```

Comment:
As discussed, the intensity layer is in float, so this increment is problematic. Switching to using floats and actual age in seconds

Reply:
What he said